src/miner.cpp

```diff
@@ -338,82 +359,87 @@ CBlockTemplate* CreateNewBlock(const CChainParams& chainparams, const MinerAddre
         // This vector will be sorted into a priority queue:
         vector<TxPriority> vecPriority;
         vecPriority.reserve(mempool.mapTx.size());
-        for (CTxMemPool::indexed_transaction_set::iterator mi = mempool.mapTx.begin();
-             mi != mempool.mapTx.end(); ++mi)
-        {
-            const CTransaction& tx = mi->GetTx();
 
-            int64_t nLockTimeCutoff = (STANDARD_LOCKTIME_VERIFY_FLAGS & LOCKTIME_MEDIAN_TIME_PAST)
-                                    ? nMedianTimePast
-                                    : pblock->GetBlockTime();
+        // If we're given a coinbase tx, it's been precomputed, so its fees are zero,
+        // so we can't include any mempool transactions; this will be an empty block.
+        if (!next_coinbase_mtx) {
```

Comment:
I'm not keen on having `zcashd` produce empty blocks by default. But I'm _less_ keen to have it burn fees by not claiming them. I guess in practice the empty-block case will only trigger when a new block arrives, after which the caller will re-poll, and on average that next call will time out on block-wait.

Reply:
@str4d all (good) mining pools do this already but it's more difficult and error prone.

Reply:
> But I'm _less_ keen to have it burn fees by not claiming them.

I briefly thought about that very idea: When a new block arrives, why not return a template that includes the transactions in the mempool -- _even though_ we're also going to include the zero-fee precalculated coinbase in that template. Burning the fees, as you said. I wondered if there's any reason _not_ to. It would just mean that whoever solves the _next_ block wouldn't be able to include those transactions (since they're already in a confirmed block). 

But there is a reason not to do that: _This very miner_ (or mining pool) may solve the next block. Then you'd surely want to collect those fees.